apps/mobile/src/hooks/useAnalytics.ts

```diff
@@ -0,0 +1,37 @@
+import { useEffect } from 'react'
+import { useLocalSearchParams } from 'expo-router'
+import { useAppSelector } from '@/src/store/hooks'
+import { selectActiveSafe } from '@/src/store/activeSafeSlice'
+import { setChainId, setSafeAddress } from '@/src/services/analytics'
+
+export const useAnalytics = () => {
+  const activeSafe = useAppSelector(selectActiveSafe)
+  const params = useLocalSearchParams<{
+    safeAddress?: string
+    chainId?: string
+  }>()
+
+  // Determine which safe to use - route params override activeSafe
+  const currentSafeAddress = params.safeAddress || activeSafe?.address
+  const currentChainId = params.chainId || activeSafe?.chainId
+
+  // Set chain ID when it changes
+  useEffect(() => {
+    if (currentChainId) {
+      setChainId(currentChainId)
```

Comment:
Isn't it good to also track when the user has changed the chainId and wallet address as well?